### PR TITLE
Small refactor on ImageComparison utils.

### PR DESCRIPTION
**Pull request by hershey5045**
Small refactor to pull out function. 

Linked to this ASV PR: https://github.com/o3de/o3de-atom-sampleviewer/pull/325

Signed-off-by: hershey5045 <43485729+hershey5045@users.noreply.github.com>

### CODE REVIEW

File: Gems/Atom/Utils/Code/Source/ImageComparison.cpp [hunk 1]
```diff
@@ -65,14 +75,7 @@ namespace AZ
...
             {
-                // We use the max error from a single channel instead of accumulating the error from each channel.
-                // This normalizes differences so that for example black vs red has the same weight as black vs yellow.
-                const int16_t diffR = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[i]) - aznumeric_cast<int16_t>(bufferB[i])));
-                const int16_t diffG = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[i + 1]) - aznumeric_cast<int16_t>(bufferB[i + 1])));
-                const int16_t diffB = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[i + 2]) - aznumeric_cast<int16_t>(bufferB[i + 2])));
-                const int16_t maxDiff = AZ::GetMax(AZ::GetMax(diffR, diffG), diffB);
-
-                const float finalDiffNormalized = maxDiff / 255.0f;
+                const float finalDiffNormalized = CalcMaxChannelDifference(bufferA, bufferB, i) / 255.0f;
```
Comment (rgba16f) [inline Gems/Atom/Utils/Code/Source/ImageComparison.cpp:78]: nit: might want to cast the MaxChannelDiff up to a float.

File: Gems/Atom/Utils/Code/Source/ImageComparison.cpp [hunk 2]
```diff
@@ -12,6 +12,16 @@ namespace AZ
 {
     namespace Utils
     {
+        int16_t CalcMaxChannelDifference(AZStd::array_view<uint8_t> bufferA, AZStd::array_view<uint8_t> bufferB, size_t index)
+        {
+            // We use the max error from a single channel instead of accumulating the error from each channel.
+            // This normalizes differences so that for example black vs red has the same weight as black vs yellow.
+            const int16_t diffR = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index]) - aznumeric_cast<int16_t>(bufferB[index])));
+            const int16_t diffG = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index + 1]) - aznumeric_cast<int16_t>(bufferB[index + 1])));
+            const int16_t diffB = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index + 2]) - aznumeric_cast<int16_t>(bufferB[index + 2])));
```
Comment (rgba16f) [inline Gems/Atom/Utils/Code/Source/ImageComparison.cpp:21]: Does abs change the result type? is the outer static cast needed?

**Review by rgba16f — APPROVED** [review]

File: Gems/Atom/Utils/Code/Source/ImageComparison.cpp (hunk 2 above)
Comment (hershey5045) [reply Gems/Atom/Utils/Code/Source/ImageComparison.cpp:21]: Yes it's needed, abs returns an int.

File: Gems/Atom/Utils/Code/Source/ImageComparison.cpp (hunk 1 above)
Comment (hershey5045) [reply Gems/Atom/Utils/Code/Source/ImageComparison.cpp]: cast added!

**Review by santorac — APPROVED** [review]

**Review by spham-amzn — COMMENTED** [review]
Now that it has been refactored out, it looks like a nice unit test candidate

**Review by spham-amzn — APPROVED** [review]

**Pull request merged**